packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -0,0 +1,408 @@
+    let style: any = {};
+
+    switch (this.props.labelPosition) {
+      case Position.start:
+        style = { float: 'left', marginRight: this.props.labelGapSpace + 'px' };
+        break;
+      case Position.end:
+        style = { float: 'right', marginLeft: this.props.labelGapSpace + 'px' };
+        break;
+      case Position.top:
+        style = { marginBottom: this.props.labelGapSpace + 'px' };
+        break;
+      case Position.bottom:
+        style = { marginTop: this.props.labelGapSpace + 'px' };
+    }
+
+    return style;
+  }
+
+  private _onChange() {
+    /**
+     * A noop input change handler.
+     * https://github.com/facebook/react/issues/7027.
+     * Using the native onInput handler fixes the issue but onChange
+     * still need to be wired to avoid React console errors
+     * TODO: Check if issue is resolved when React 16 is available.
+     */
+  }
+
+  /**
+   * This is used when validating text entry
+   * in the input (not when changed via the buttons)
+   * @param event - the event that fired
+   */
+  @autobind
+  private _validate(event: React.FocusEvent<HTMLInputElement>) {
+    const element: HTMLInputElement = event.target as HTMLInputElement;
+    const value: string = element.value;
+    if (this.state.value) {
+      const newValue = this._onValidate(value);
+      if (newValue) {
+        this._lastValidValue = newValue;
+        this.setState({ value: newValue });
+      }
+    }
+  }
+
+  /**
+   * The method is needed to ensure we are updating the actual input value.
+   * without this our value will never change (and validation will not have the correct number)
+   * @param event - the event that was fired
+   */
+  @autobind
+  private _onInputChange(event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>): void {
+    const element: HTMLInputElement = event.target as HTMLInputElement;
+    const value: string = element.value;
+
+    this.setState({
+      value: value,
+    });
+  }
+
+  /**
+   * Update the value with the given stepFunction
+   * @param shouldSpin - should we fire off another updateValue when we are done here? This should be true
+   * when spinning in response to a mouseDown
+   * @param stepFunction - function to use to step by
+   */
+  @autobind
+  private _updateValue(shouldSpin: boolean, stepFunction: (string) => string | void) {
+    const newValue = stepFunction(this.state.value);
+    if (newValue) {
+      this._lastValidValue = newValue;
+      this.setState({ value: newValue });
+    }
+
+    if (this._spinningByMouse !== shouldSpin) {
+      this._spinningByMouse = shouldSpin;
+    }
+
+    if (shouldSpin) {
+      this._currentStepFunctionHandle = this._async.setTimeout(() => { this._updateValue(shouldSpin, stepFunction); }, this._stepDelay);
+    }
+  }
+
+  /**
+   * Stop spinning (clear any currently pending update and set spinning to false)
+   */
+  @autobind
+  private _stop() {
+    if (this._currentStepFunctionHandle !== null) {
```

Comment:
_currentStepFunctionHandle is a number, should this be explicitly comparing to 0 instead?